sql/src/test/java/com/rilldata/calcite/CalciteToolboxTest.java

```diff
@@ -0,0 +1,123 @@
+  {
+    List<MigrationStep> migrationSteps = CalciteToolbox.inferMigrations(
+        """
+            create view a as select 'a' as d, 2 as m ;
+            create metrics view b dimensions d measures count(m) from a
+        """,
+        """
+          {
+            "entities": [
+              {
+                "name": "b",
+                "columns": [
+                  {
+                    "name": "a",
+                    "type": "int"
+                  }
+                ],
+                "ddl": "create table b (a int)"
+              }
+            ]
+          }
+        """, PostgresqlSqlDialect.DEFAULT);
+    Assertions.assertEquals("DROP TABLE \"B\"", migrationSteps.get(0).ddl);
+    Assertions.assertEquals("CREATE VIEW \"A\" AS\nSELECT 'a' AS \"D\", 2 AS \"M\"", migrationSteps.get(1).ddl);
+    Assertions.assertEquals(2, migrationSteps.size());
+  }
+
+  @Test
+  public void testMigrationsWithFullName() throws SqlParseException, JsonProcessingException
+  {
+    List<MigrationStep> migrationSteps = CalciteToolbox.inferMigrations("create view a as select 1", """
+          {
+            "entities": [
+              {
+                "name": "b",
+                "columns": [
+                  {
+                    "name": "a",
+                    "type": "int"
+                  }
+                ],
+                "ddl": "create table main.b (a int)"
+              }
+            ]
+          }
+          """, PostgresqlSqlDialect.DEFAULT);
+    Assertions.assertEquals("DROP TABLE \"MAIN\".\"B\"", migrationSteps.get(0).ddl);
+    Assertions.assertEquals("CREATE VIEW \"A\" AS\nSELECT 1", migrationSteps.get(1).ddl);
+  }
+
+  @Test
+  public void testMultipleMigrations() throws SqlParseException, JsonProcessingException
+  {
+    List<MigrationStep> migrationSteps = CalciteToolbox.inferMigrations("create table a (id int) ; create view b as select * from a", """
+          {
+            "entities": [
+              {
+                "name": "a",
+                "columns": [
+                  {
+                    "name": "id",
+                    "type": "int"
+                  }
+                ],
+                "ddl": "create table a (id int)"
+              },
+              {
+                "name": "b",
+                "columns": [
+                  {
+                    "name": "id",
+                    "type": "int"
+                  }
+                ],
+                "ddl": "create view b as select * from a"
+              }
+            ]
+          }
+          """, PostgresqlSqlDialect.DEFAULT);
+    Assertions.assertEquals("DROP TABLE \"A\"", migrationSteps.get(0).ddl);
+    Assertions.assertEquals("DROP VIEW \"B\"", migrationSteps.get(1).ddl);
+    Assertions.assertEquals("CREATE TABLE \"A\" (\"ID\" INTEGER)", migrationSteps.get(2).ddl);
+    Assertions.assertEquals("CREATE VIEW \"B\" AS\nSELECT *\nFROM \"A\"", migrationSteps.get(3).ddl);
```

Comment:
Given that the input ddl and the catalog ddl is equivalent here, shouldn't it do no migrations?

Also, disregarding the above, shouldn't it drop in reverse DAG order – i.e. first drop the `B` view, then drop the `A` table (since `B` relies on `A`)?